File: adventofcode/Day6.js
var fs = require('fs');
var readline = require('readline');

var readable = fs.createReadStream(__dirname + '/test.in', {
    encoding: 'utf8'
});
var interface = readline.createInterface({
  input: readable,//process.stdin
  terminal: false
});

var m = Array(1000).fill().map(()=>Array(1000).fill().map(()=>0));
interface.on('line', function(line){
    var s = line.split(' ');
    if(s[0] === 'turn'){
        var val = -1
        var a = s[2].split(',').map((x)=>+x);
        var b = s[4].split(',').map((x)=>+x);
        if(s[1] === 'on')
            val = 1
        for(var i=a[0]; i<=b[0]; i++){
            for(var j=a[1]; j<=b[1]; j++){
                m[i][j] += val;
                if(m[i][j] < 0)
                    m[i][j] = 0;
            }
        }
    } else {
        var a = s[1].split(',').map((x)=>+x);
        var b = s[3].split(',').map((x)=>+x);
        for(var i=a[0]; i<=b[0]; i++)
            for(var j=a[1]; j<=b[1]; j++)
                m[i][j] += 2;
    }
});

interface.on('close', function(){
    console.log(m.reduce((a,b)=>a + b.reduce((b1,b2)=>b1 + b2),0));
});

File: adventofcode/Day6.py
def solve():
    m = [[0 for j in range(1000)] for i in range(1000)];
    for i in range(300):
        s = interface.r().split(' ')
        if s[0] == 'turn':
            val = -1
            a = [int(i) for i in s[2].split(',')]
            b = [int(i) for i in s[4].split(',')]
            if s[1] == 'on':
                val = 1
            for i in range(a[0], b[0]+1):
                for j in range(a[1], b[1]+1):
                    m[i][j] += val
                    if m[i][j] < 0:
                        m[i][j] = 0
        else:
            a = [int(i) for i in s[1].split(',')]
            b = [int(i) for i in s[3].split(',')]
            for i in range(a[0], b[0]+1):
                for j in range(a[1], b[1]+1):
                    m[i][j] += 2
    interface.w(sum([sum(x) for x in m]))

class interface:
    @staticmethod
    def w(*s):
        #fout.write(s + "\n")
        print(*s)
    @staticmethod
    def r():
        return fin.readline().rstrip()
        #return input()

fin = open('test.in', 'r')
solve()
fin.close()
